index.js

```diff
@@ -499,20 +499,16 @@ function signAsync (opts) {
     })
     .then(function () {
       // Pre-sign operations
-      var promise = Promise.resolve()
-      if (opts.version ? compareVersion(opts.version, '1.1.1') >= 0 : true) {
+      if (opts.entitlements != null && (opts.version == null || compareVersion(opts.version, '1.1.1') >= 0)) {
         // Enable Mac App Store sandboxing without using temporary-exception, introduced in Electron v1.1.1. Relates to electron#5601
         if (opts['pre-auto-entitlements'] === false) {
           debugwarn('Pre-sign operation disabled for entitlements automation.')
         } else {
           debuglog('Pre-sign operation enabled for entitlements automation with versions >= `1.1.1`; disable by setting `pre-auto-entitlements` to `false`.')
-          promise = promise.then(function () {
-            debuglog('Automating entitlement app group...')
-            return preAutoEntitlementAppGroupAsync(opts)
-          })
+          debuglog('Automating entitlement app group...')
+          return preAutoEntitlementAppGroupAsync(opts)
         }
       }
-      return promise
```

Comment:
No, 

>  If the value is a thenable (i.e. has a then method), the returned promise will "follow" that thenable, adopting its eventual state; otherwise the returned promise will be fulfilled with the value. 

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise

If you want to state that `return` is not missed, return null: `return null`.